torch/distributed/_spmd/distribute.py

```diff
@@ -159,6 +169,35 @@ def _remap_arg(node_to_obj: Dict[fx.Node, object], arg: object) -> object:
         return arg
 
 
+def binop_sym_int_consumer_rule(node: fx.Node, args: Tuple[object, ...]) -> DTensor:
+    assert len(args) == 2, f"Expect two args but got op {node.target} with args {args}"
+    assert isinstance(
+        args[0], DTensor
+    ), f"Expect 1st argument to be DTensor but got {args[0]}"
+    assert isinstance(args[1], list), f"Expect 2nd argument as list but got {args[1]}"
+
+    local_sizes = [a.size if isinstance(a, DSize) else a for a in args[1]]
+    placements = [a.placement if isinstance(a, DSize) else Replicate() for a in args[1]]
+    assert len([p for p in placements if p.is_shard()]) == args[0].device_mesh.ndim
+
+    # set node args to real int sizes.
+    node.args = (node.args[0], local_sizes)
+    op = cast(torch._ops.OpOverload, node.target)
+    return DTensor.from_local(
+        local_tensor=op(args[0]._local_tensor, local_sizes),
+        device_mesh=args[0].device_mesh,
+        placements=[Shard(i) for i, p in enumerate(placements) if p.is_shard()],
+        run_check=False,
+    )
+
+
+# Dispatch override for
+SYM_INT_CONSUMERS: Dict[torch._ops.OpOverload, Callable] = {
```

Comment:
would this approach also works for the tensor factory method (i.e. full, arange)?

Reply:
yep, I assume so.